behind-the-veil-siteroot/imports/ui/components/searchBar/searchBar.jsx

```diff
@@ -4,57 +4,244 @@
+                setFilteredSuggestions([]);
+                setSelectedSuggestions(false) // reset state
+                handleButtonClickOrSubmit()
+            } else {
+                // If a suggestion has NOT been selected, filter suggestions based on input value
+                const filteredSuggestions = fullSuggestions.filter(suggestion => {
+                    // filter suggestions based on if it matches either main or sub criteria
+                    const matchMain = suggestion.main.toLowerCase().includes(inputValue.toLowerCase());
+                    const matchSub = suggestion.sub ? suggestion.sub.toLowerCase().includes(inputValue.toLowerCase()) : false;
+                    return matchMain || matchSub;
+                });
+                setFilteredSuggestions(filteredSuggestions);
+            }
+        } else {
+            setFilteredSuggestions([]);
+        }
+    }, [inputValue])
+
+    // on search type change
+    useEffect(() => {
+        setFullSuggestions(useSearchSuggestions(searchType, usersData, servicesData)
+            .sort((a, b) => a.main.localeCompare(b.main)));
+        setFilteredSuggestions([]);
+    }, [searchType])
+
+
+    // search type input handler
+    const handleSearchTypeChange = (event) => {
+        setSearchType(event.target.value);
+    };
 
     // Function that resets the input value of the search bar
     const handleReset = () => {
-        onChange({target: {value: ''}});
+        setInputValue('');
+        setFilteredSuggestions([]);
     };
 
-    return (
-        // The form which holds the search input field and the reset button
-        <form className="flex h-12" onSubmit={handleSubmit}>
-            {/* The search input field (i.e. the search bar) */}
-            <Input
-                {...searchBarProps}
-                type="search"
-                className={classNames("rounded-r-none border-r-0 w-[250px] sm:w-[35vw] ", className)}
-                placeholder={placeholderName}
-                value={value}
-                onChange={onChange}
-            />
-            {/* The reset button which resets the value of the search input field to an empty string */}
-            <button type="button"
-                    className={"input-base flex justify-center items-center w-12 border-l-0 rounded-l-none"}
-                    onClick={handleReset}>
-                <XMarkIcon className="size-7 stroke-2 rounded-full p-1
-                hover:bg-white-hover active:bg-light-grey-hover transition duration-200 ease-in-out"></XMarkIcon>
-            </button>
-        </form>
-    );
+    // when a suggestion is selected, use the value of the suggestion as the input, to trigger the use effect that trigger
+    // on input change above
+    const handleSuggestionSelect = (value) => {
+        setInputValue(value.main);
+        setSelectedSuggestions(true) // this ensures code goes through a different logic branch than for normal input change
```

Comment:
I think selectedSuggestions is an unnecessary state. Instead of setting a state here so that the above useEffect callback function runs a particular set of logic, you can probably just run the logic here. Triggering the useEffect and using a state to then ensure only particular logic in that function runs is quite difficult to follow and also more likely to cause errors.

Reply:
removed state